src/smt/sygus_solver.cpp

```diff
@@ -233,39 +271,47 @@ Result SygusSolver::checkSynth(Assertions& as)
   return r;
 }
 
-bool SygusSolver::getSynthSolutions(std::map<Node, Node>& sol_map)
+bool SygusSolver::getSynthSolutions(std::map<Node, Node>& solMap)
 {
   Trace("smt") << "SygusSolver::getSynthSolutions" << std::endl;
-  std::map<Node, std::map<Node, Node>> sol_mapn;
+  if (usingSygusSubsolver())
+  {
+    // use the call to get the synth solutions from the subsolver
+    return d_subsolver->getSubsolverSynthSolutions(solMap);
+  }
```

Comment:
if you have this why all the upstream methods to get subsolver solutions?

Reply:
do you mean why are other methods using the `SolverEngine::getSubsolverSynthSolutions` interface?

Reply:
right, like the one in SMT engine

Reply:
`usingSygusSubsolver` is true if we use a sygus subsolver if we received `check-synth` calls.  The differences is that a subsolver may receive a synthesis conjecture and must return synth solutions after `check-sat` calls.

Reply:
The documentation https://github.com/cvc5/cvc5/pull/7736/files#diff-f7beb6dbdfd17c4aa0317171df9c685a78d8e393e5a4fc0137ba87d1c3f890e5R567 makes this distinction.